### ISSUE TITLE
response with Exception not cleared out of `_responses`

### ISSUE THREAD

**Issue by MattBroach** [pnuckowski/aioresponses#86]
I have an API Client I've written that uses `aiohttp` under the hood access various APIs.  Amongst its tasks are making sure it handles errors/bad responses from the API well without bringing the program to a screaming halt, and one such problem is handling aiohttp's `ClientPayloadError` exception -- that it continues to retry the endpoint while logging the response.  I wrote a test to make sure this works properly:

```python
    @aioresponses()
    def test_ClientPayloadError_is_handled_properly(self, mocked):
        """
        a ClientPayloadError should log that it occurred, and then retry the same link
        """
        mocked.get(
            'https://test.url/objects/10/', exception=ClientPayloadError('oops'),
        )
        mocked.get(
            'https://test.url/objects/10/', payload={'test': 'data'}, status=200, exception=None,
        )

        with self.assertRaises(APIException) as e:
            with self.assertLogs() as logs:
                data = self.loop.run_until_complete(self.api_client.get_async('objects/10/'))

        self.assertEqual(data, {'test': 'data'})
        self.assertEqual(logs.output, ([
              'expected log output blah blah'
        ]))
```

Unfortunately, when the test above is run, the `ClientPayloadError` response is repeatedly raised until the retry limit is raised, rather than running the second (non-Exception) response. 

My suspicion is that the `_responses` array is not being updated in `match`, since the exception stops the code that removes the URlResponse from `_responses` from running, but I haven't had a chance to really dig into it.  I would, however, be happy to open a PR with a failing test, if it would be helpful for you.

**Comment by pnuckowski** [comment]
Thanks for reporting the issue with such explanation and test example :) It will help me a lot :)

Should be fixed soon.

**Comment by pnuckowski** [comment]
Close issue. Fix will be in version 0.4.0
